pkg/network/ovn_kubernetes.go

```diff
@@ -415,35 +428,77 @@ func bootstrapOVNHyperShiftConfig(hc *HyperShiftConfig, kubeClient cnoclient.Cli
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.Infof("Did not find ovnkube-sbdb route")
-		} else {
-			return nil, fmt.Errorf("Could not get ovnkube-sbdb route: %v", err)
+	for _, svc := range hcp.Spec.Services {
+		// TODO: instead of the hardcoded string use ServiceType hyperv1.OVNSbDb once the API is updated
+		if svc.Service == "OVNSbDb" {
+			ovnHypershiftResult.ServicePublishingStrategy = &svc.ServicePublishingStrategy
 		}
-	} else {
-		err := runtime.DefaultUnstructuredConverter.FromUnstructured(routeObj.UnstructuredContent(), route)
-		if err != nil {
-			return ovnHypershiftResult, err
+	}
+	if ovnHypershiftResult.ServicePublishingStrategy == nil {
+		klog.Warningf("service publishing strategy for OVN southbound database does not exist in hyperv1.HostedControlPlane %s/%s. Defaulting to route", hc.Name, hc.Namespace)
+		ovnHypershiftResult.ServicePublishingStrategy = &hyperv1.ServicePublishingStrategy{
+			Type: hyperv1.Route,
 		}
-		if route.Spec.Host == "" && route.Status.Ingress[0].Host == "" {
-			return ovnHypershiftResult, nil
+	}
+	switch ovnHypershiftResult.ServicePublishingStrategy.Type {
+	case hyperv1.Route:
+		{
+			route := &routev1.Route{}
+			gvr := schema.GroupVersionResource{
+				Group:    "route.openshift.io",
+				Version:  "v1",
+				Resource: "routes",
+			}
+			clusterClient := kubeClient.ClientFor(client.ManagementClusterName)
+			routeObj, err := clusterClient.Dynamic().Resource(gvr).Namespace(hc.Namespace).Get(context.TODO(), "ovnkube-sbdb", metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					klog.Infof("Did not find ovnkube-sbdb route")
+				} else {
+					return nil, fmt.Errorf("could not get ovnkube-sbdb route: %v", err)
+				}
+			} else {
+				err := runtime.DefaultUnstructuredConverter.FromUnstructured(routeObj.UnstructuredContent(), route)
+				if err != nil {
+					return ovnHypershiftResult, err
+				}
+				if (len(route.Status.Ingress) < 1 || route.Status.Ingress[0].Host == "") && route.Spec.Host == "" {
+					return ovnHypershiftResult, nil
+				}
+				if len(route.Status.Ingress) >= 1 && route.Status.Ingress[0].Host != "" {
+					ovnHypershiftResult.OVNSbDbEndpoint = fmt.Sprintf("ssl:%s:%s", route.Status.Ingress[0].Host, OVN_SB_ROUTE_PORT)
+				} else if route.Spec.Host != "" {
+					ovnHypershiftResult.OVNSbDbEndpoint = fmt.Sprintf("ssl:%s:%s", route.Spec.Host, OVN_SB_ROUTE_PORT)
+				}
+				klog.Infof("Overriding OVN configuration route to %s", ovnHypershiftResult.OVNSbDbEndpoint)
+			}
 		}
-		if route.Status.Ingress[0].Host != "" {
-			ovnHypershiftResult.Route = route.Status.Ingress[0].Host
-		} else if route.Spec.Host != "" {
-			ovnHypershiftResult.Route = route.Spec.Host
+	case hyperv1.NodePort:
+		{
+			svc := &corev1.Service{}
+			err = kubeClient.Default().CRClient().Get(context.TODO(), types.NamespacedName{Namespace: hc.Namespace, Name: "ovnkube-master"}, svc)
```

Comment:
should this be ovnkube-master-external service?

Reply:
yes! done